package.json

```diff
@@ -2,61 +2,77 @@
   "name": "bind-event-listener",
   "version": "3.0.0",
   "private": false,
-  "description": "Making binding and unbinding DOM events easier",
+  "type": "module",
+  "description": "Safe DOM Event binding and unbinding",
   "author": "Alex Reardon <alexreardon@gmail.com>",
   "license": "MIT",
   "repository": {
     "type": "git",
     "url": "https://github.com/alexreardon/bind-event-listener.git"
   },
   "keywords": [
+    "event listener",
+    "event handler",
+    "dom events",
     "dom",
     "ui",
-    "event listener",
-    "event handler"
+    "bind"
   ],
-  "main": "dist/index.js",
-  "types": "dist/index.d.ts",
   "sideEffects": false,
+  "main": "./dist/cjs/index.js",
+  "module": "./dist/esm/index.js",
+  "types": "./dist/esm/index.d.ts",
```

Comment:
TS will definitely complain about this with `moduleResolution: node16` in many situations (when the importing file will be assumed to be targeting CJS - which is pretty common). TS tries to reflect the node's runtime and right now you made this an error: `require('bind-event-listener')` in all modern node versions (12+)

Reply:
That's slightly complicated and why I recommend using a lib builder. You don't want to implement this house of cards urself ;p

Preconstruct generates:
- CJS files
- MJS wrappers for those (ESM files that import the CJS files) to preserve strict module shape for node consumers who might *import* this package
- ESM files that can directly be used by bundlers (pointed to by `package.json#module` **and** `package.json#exports.module`)

This is the combination that I find the most correct and the least painful. Since this package likely doesn't suffer from dual package hazard u could also get away with using `tsup` or similar and ship separate CJS and ESM files. This risks your library being loaded twice (once by the require loader and once by the import loader) but I don't think it's that big of a deal here. That said, Preconstruct's strategy is optimized to avoid this altogether.

Reply:
Interesting. I'll take a look a preconstruct

Reply:
So if I go esm only, then consumers won't be able to do `require('bind-event-listener')`?

Reply:
It looks like pre-construct sets `"main"` as the cjs and "exports" as the esm (without a `"type": "module"`)

https://preconstruct.tools/tutorials/#what-did-we-just-do

Reply:
with Preconstruct's strategy you shouldn't set `type: module` at all. You want to enable those settings:
https://github.com/Andarist/react-textarea-autosize/blob/5953e024310ab0b6ad1c26b2bc36addb9cc23afa/package.json#L140-L147

and then run `preconstruct fix`

Reply:
Since the package is in TS and Preconstruct is still Babel-based... you might have to use the typescript preset, like here:
https://github.com/bvaughn/react-resizable-panels/blob/9c72136e7ffd399e28c4196604fc86940bfa04b3/babel.config.js

This is an example PR that switches to using Preconstruct: https://github.com/bvaughn/react-error-boundary/pull/144

Reply:
Seems like a lot of hoops. Very tempted to go esm only. Or maybe see if I can get away with a main field that is cjs 🤔